src/main/java/degree/CatalogChecker.java

```diff
@@ -0,0 +1,290 @@
+package degree;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.microsoft.azure.functions.ExecutionContext;
+
+import FuncTest.TestFunc.globals;
+import homework.LoginCredentials;
+
+public class CatalogChecker {
+	
+	ExecutionContext c;
+	List<Course> myCourses;
+	List<Course> coreCourses;
+	
+	public CatalogChecker(ExecutionContext c, LoginCredentials creds) {
+		this.c = c;
+		myCourses = (new CourseListGetter(creds, c)).getCourseList();
+		
+		if(myCourses.size() > 0) {
+			c.getLogger().info("================== GOT YOUR COURSES ====================");
+		} else {
+			c.getLogger().info("================== GOT YOUR COURSES ====================");
+			throw new RuntimeException("Failed to read course from UG\n");
+		}
+	}
+	
+	static public double sumPoints(final List<Course> list){
+		return list.stream().map(c -> Double.valueOf(c.getPoints()))
+				.reduce(Double.valueOf(0.0), (x,y) -> Double.valueOf(x.doubleValue()+y.doubleValue())).doubleValue();
+	}
+	
+	public List<Course> getMissingMandatory() throws SQLException{
+		List<Course> mandatory = getMandatory(), diff = difference(mandatory, myCourses);
+		if(myCourses.contains(new Course(234145, -1)))
+			diff.remove(new Course(44145, -1));
+		if(myCourses.contains(new Course(44145, -1)))
+			diff.remove(new Course(234145, -1));
+		return diff;
+	}
+	
+	public String degreeCompletionCompute() throws SQLException {
+		StringBuilder jsonResult = new StringBuilder();
+		boolean finished = true;
+		
+		List<Course> missingMandatory = getMissingMandatory();
+		jsonResult.append("Category - Mandatory Classes:\n");
+		if (missingMandatory.isEmpty())
+			jsonResult.append("You've completed the mandatory courses\n");
+		else {
+			jsonResult.append("It seems that you haven't completed the following mandatory courses:\n");
+			for (Course course : missingMandatory)
+				jsonResult.append(course.getCourseNum() + " ");
```

Comment:
Do you have the names of the courses? If so, please consider printing the names instead of the numbers, or maybe both

Reply:
The courses a user did are parsed from their score sheet which is only available in Hebrew, so for now - for this POC it only shows numbers.